util/snapshot/src/lib.rs

```diff
@@ -73,6 +73,18 @@ impl Snapshot {
         }
     }
 
+    pub fn refresh(&self, store: StoreSnapshot) -> Snapshot {
```

Comment:
Why we need `refresh`? When should we `refresh`? Please add some comment here to explain it.